### PR TITLE
Bug/26009 hpa pupil register page is throwing 500

### PR DESCRIPTION
**Pull request by jon-shipley**
* Prevent error being thrown by ensuring var is defined
* Backfill check started date with audit data if the data is not yet there when the completed check message is received
* prevent duplicate check submissions

### CODE REVIEW

File: functions/completed-checks/index.js [hunk 1]
```diff
@@ -79,6 +79,25 @@ async function savePayloadToAdminDatabase (completedCheckMessage, logger) {
...
 
+  // Don't process any checks more than once
+  if (check.receivedByServerAt) {
+    const msg = `completed-check: ERROR: payload re-submission is banned for check ${checkData.checkCode}`
+    logger.error(msg)
+    throw new Error(msg)
+  }
+
+
+  if (!checkData.startedAt) {
```
Comment (GeorgeChatzigiannis) [inline functions/completed-checks/index.js]: @jon-shipley under which circumstances the `startedAt` field would not be populated?

**Comment by GeorgeChatzigiannis** [comment]
@jon-shipley looks good just 1 comment

File: functions/completed-checks/index.js (hunk 1 above)
Comment (GeorgeChatzigiannis) [reply functions/completed-checks/index.js]: is it only in case check-started function fails?

Comment (jon-shipley) [reply functions/completed-checks/index.js]: @GeorgeChatzigiannis 

1.  The client browser had networking issues
2. The completed check message was received before the check started message (busy system?)

Comment (GeorgeChatzigiannis) [reply functions/completed-checks/index.js]: Ok that makes sense to me, thanks.

**Pull request merged**